pipenv/cli/command.py

```diff
@@ -760,20 +761,17 @@ def requirements(state, dev=False, dev_only=False, hash=False):
     for i, package_index in enumerate(lockfile["_meta"]["sources"]):
         prefix = "-i" if i == 0 else "--extra-index-url"
         echo(crayons.normal(" ".join([prefix, package_index["url"]])))
+
+    deps = {}
     if not dev_only:
-        for req_name, value in lockfile["default"].items():
-            if hash:
-                hashes = [f" \\\n    --hash={h}" for h in value.get("hashes", [])]
-            else:
-                hashes = []
-            echo(crayons.normal("".join([req_name, value["version"], *hashes])))
+        deps.update(lockfile['default'])
     if dev or dev_only:
-        for req_name, value in lockfile["develop"].items():
-            if hash:
-                hashes = [f" \\\n    --hash={h}" for h in value.get("hashes", [])]
-            else:
-                hashes = []
-            echo(crayons.normal("".join([req_name, value["version"], *hashes])))
+        deps.update(lockfile['develop'])
+
+    pip_deps = convert_deps_to_pip(deps, r=False, include_index=False, with_hashes=hash, with_markers=False)
```

Comment:
I also tried like PR [#5071](https://github.com/pypa/pipenv/pull/5071) to modify it, but there is a much better code.
I will try my best to make a great code like you. 😎
But there is an error in this part. It needs to be modified.
```
pip_deps = convert_deps_to_pip(deps, r=False, include_index=False, include_hashes=hash, include_markers=False)
```